Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+ToolbarDelegate.swift

```diff
@@ -926,6 +913,51 @@ extension BrowserViewController: ToolbarDelegate {
 
 extension BrowserViewController: UIContextMenuInteractionDelegate {
   public func contextMenuInteraction(_ interaction: UIContextMenuInteraction, configurationForMenuAtLocation location: CGPoint) -> UIContextMenuConfiguration? {
+    
+    var configuration: UIContextMenuConfiguration?
+    
+    switch interaction.view?.accessibilityIdentifier {
+    case topToolbar.locationView.reloadButtonAccessibilityIdentifier:
+      configuration = reloadButtonMenuConfiguration
+    default:
+      configuration = locationViewMenuConfiguration
+    }
+    
+    if #available(iOS 16.0, *) {
+      configuration?.preferredMenuElementOrder = .priority
+    }
+    
+    return configuration
+  }
+  
+  private var reloadButtonMenuConfiguration: UIContextMenuConfiguration {
+    let configuration =  UIContextMenuConfiguration(identifier: nil, previewProvider: nil) { [unowned self] _ in
+      var actionMenu: [UIMenu] = []
+      
+      let tab = tabManager.selectedTab
+      
+      let title = tab?.isDesktopSite == true ? Strings.appMenuViewMobileSiteTitleString : Strings.appMenuViewDesktopSiteTitleString
+      
+      let icon = tab?.isDesktopSite == true ? "leo.smartphone" : "leo.monitor"
+      
+      let copyAction = UIAction(
+        title: title,
+        image: UIImage(braveSystemNamed: icon),
+        handler: UIAction.deferredActionHandler { [weak tab] _ in
+          tab?.switchUserAgent()
+        })
+
+      let copyMenu = UIMenu(title: "", options: .displayInline, children: [copyAction])
+      
+      actionMenu = [copyMenu]
+      
+      return UIMenu(title: "", identifier: nil, children: actionMenu)
```

Comment:
Action Menu is not edited, it can be removed and action can be added here direct

return UIMenu(title: "",  identifier: nil, children:  [copyMenu])